core/src/main/java/org/apache/iceberg/MetadataColumns.java

```diff
@@ -152,4 +154,19 @@ public static boolean isMetadataColumn(int id) {
   public static boolean nonMetadataColumn(String name) {
     return !isMetadataColumn(name);
   }
+
+  public static Schema schemaWithRowLineage(Table table) {
+    Preconditions.checkArgument(null != table, "Invalid table: null");
+    Preconditions.checkArgument(
+        !(table instanceof BaseMetadataTable),
+        "Cannot produce row lineage for metadata table: %s",
+        table);
+    Schema rowLineageSchema =
+        new Schema(
+            MetadataColumns.metadataColumn(table, MetadataColumns.ROW_ID.name()).asOptional(),
+            MetadataColumns.metadataColumn(
+                    table, MetadataColumns.LAST_UPDATED_SEQUENCE_NUMBER.name())
+                .asOptional());
+    return TypeUtil.join(table.schema(), rowLineageSchema);
```

Comment:
Looks like this is constructing `Schema` instances just to satisfy the types expected by `TypeUtil.join`. Is there an alternative method? Should there be?

Reply:
I think using TypeUtil.join as it is today is right. My mental model is the row lineage fields are a separate schema themselves that we combine with normal schemas when we need to. 

Alternatively we could have an API that joins vararg fields? (`join(Types.NestedField... fields)`) that would avoid clients creating the schema

Reply:
The update looks good. We may want to have a few different forms, like `join(Schema, NestedField...)` but this is much simpler now that the `Table` has been removed. Thanks!